src/include/duckdb/common/types/vector.hpp

```diff
@@ -204,6 +205,8 @@ struct ConstantVector {
 		return vector.validity;
 	}
 	DUCKDB_API static const SelectionVector *ZeroSelectionVector(idx_t count, SelectionVector &owned_sel);
+	//! Turns "vector" into a constant vector by referencing a value within the other vector
+	DUCKDB_API static void Reference(Vector &vector, Vector &other, idx_t position, idx_t count);
```

Comment:
Can we use `const` here to make the input and output clearer?

Reply:
I have renamed "other" to "source", but unfortunately I cannot make the source "const" since the function might call `Orrify`, which can modify the vector. Specifically, in the case of some more exotic compression formats (SEQUENCE, and likely RLE in the future) a (partial) decompression occurs that we want to save in the vector to avoid having to repeat it later on.